lib/postgresql.js

```diff
@@ -132,15 +108,19 @@ PostgreSQL.prototype.executeSQL = function (sql, params, callback) {
       self.debug('SQL: ' + sql);
     }
   }
-  self.connection.query(sql, params, function (err, data) {
-    // if(err) console.error(err);
-    if (err && self.settings.debug) {
-      self.debug(err);
-    }
-    if (self.settings.debug && data) self.debug("%j", data);
-    // console.log(err);
-    if (log) log(sql, time);
-    callback(err ? err : null, data ? data.rows : null);
+
+  self.pg.connect(self.clientConfig, function(err, client, done) {
+    client.query(sql, params, function(err, data) {
+      // if(err) console.error(err);
+      if (err && self.settings.debug) {
+        self.debug(err);
+      }
+      if (self.settings.debug && data) self.debug("%j", data);
+      // console.log(err);
+      if (log) log(sql, time);
+      process.nextTick(done); // Release the pooled client
+      callback(err ? err : null, data ? data.rows : null);
+    });
```

Comment:
I see it now, I did not notice that line before.

Is there any particular reason why you are releasing the connection in the next tick and immediately (sync)?

Also `releaseConnection` would be a more suitable & less confusing name, since we are already using `done` as the name for the regular callback, e.g. in unit tests.